# DNA Metabarcoding Course

Welcome to the course focused on the analysis of data obtained by DNA metabarcoding (Illumina amplicon sequencing). In this course, we will use the e-INFRA CZ services provided to the scientific community in the Czech Republic. This course will take you from installing programs to creating abundance tables for phylogenetically-defined taxa, teaching you how to analyze DNA metabarcoding data using a wide range of bioinformatics tools. The data analysis process will include read quality analysis (FastQC, Perl scripts), paired-end assembly (PANDAseq), demultiplexing (fqgrep), clustering (swarm), chimera detection (usearch), auto-blast (SEED2), and phylogenetic delineation of taxa (IQtree).

The course consists of eight lessons.

## Lessons

[Lesson 1](lesson1/lesson1.md) - Installation of programs, Getting access to MetaCentrum
[Lesson 2](lesson2/lesson2.md) - Openstacks - Creating the instance
[Lesson 3](lesson3/lesson3.md) - Accessing Openstacks, Introduction to UNIX
[Lesson 4](lesson4/lesson4.md) - Installation of Linux programs and scripts
[Lesson 5](lesson5/lesson5.md) - Introduction to FastQ data
[Lesson 6](lesson6/lesson6.md) - Data analysis - processing of replicates
[Lesson 7](lesson7/lesson7.md) - Data analysis - demultiplexing, clustering
[Lesson 8](lesson8/lesson8.md) - Preparing the swarm table, Blast searches

Enjoy!
