# Lesson 5: Uploading and Inspecting Fastq Data

In this lesson, we will upload and inspect Fastq data from a study on the soil diversity of green algae in various parts of Europe.

## Upload Raw Data

1. **Login to the Virtual Machine using MobaXterm**
2. **Create directories for the two sequencing replicates**:
   ```bash
   mkdir /home/ubuntu/RUN1
   mkdir /home/ubuntu/RUN2
   ```

3. **Upload raw data of Illumina reads**: (Data downloaded in Lesson 1)
   - To `RUN1`:
     - 220902_SN1126_A_L001_AUHP-8_AdapterTrimmed_R1.fastq.gz
     - 220902_SN1126_A_L001_AUHP-8_AdapterTrimmed_R2.fastq.gz
   - To `RUN2`:
     - 221122_SN6662_A_L001_AUHP-11_AdapterTrimmed_R1.fastq.gz
     - 221122_SN6662_A_L001_AUHP-11_AdapterTrimmed_R2.fastq.gz

## Basic Work with Large Files

### Navigating to the Directory
```bash
cd /home/ubuntu/RUN1
```

### Using Pipes (|) â€“ Command Chaining

We can quickly peek into a compressed file without fully decompressing it using pipes.

### Viewing Files
- **cat**: Command to display file contents
- **zcat**: Command to decompress and display file contents

### Inspecting FASTQ Sequences
```bash
zcat 220902_SN1126_A_L001_AUHP-8_AdapterTrimmed_R1.fastq.gz | head
```

1. **The first line starts with @ and contains the sequence name**:
   - `@M01126`: The unique instrument name or identifier.
   - `1065`: The run number on the sequencing instrument.
   - `000000000-KKRG6`: The flowcell identifier.
   - `1`: The lane number on the flowcell.
   - `1101`: The tile number within the flowcell lane.
   - `8504`: The x-coordinate of the cluster within the tile.
   - `1271`: The y-coordinate of the cluster within the tile.
   - `1`: The read number (1 for the first read in a paired-end run, 2 for the second read).
   - `N`: Indicates whether the read passed quality control (Y for yes, N for no).
   - `0`: The control number (typically 0).
   - `GNACAGCT+TGATCNGT`: The index sequences (dual indices separated by a plus sign).
2. **The second line contains the sequence**.
3. **The third line is a plus sign (+)**.
4. **The fourth line contains the Phred Quality Score in ASCII Code**:
   - This score measures the quality of the identification of nucleobases generated by automated DNA sequencing. It is logarithmically related to the probability of a base call being incorrect.
   - Quality scores vary by sequencing instrument. Since 2011, Illumina uses Phred+33 encoding, where uppercase letters indicate high quality.

### Searching and Understanding Data

1. **Count the number of sequences**:
   ```bash
   zcat 220902_SN1126_A_L001_AUHP-8_AdapterTrimmed_R1.fastq.gz | wc -l | awk '{print $1/4}'
   ```
   Or count the occurrences of the instrument name at the start of lines:
   ```bash
   zcat 220902_SN1126_A_L001_AUHP-8_AdapterTrimmed_R1.fastq.gz | grep "^@M01126" | wc -l
   ```

2. **View sequence names** (`q` to quit `more`):
   ```bash
   zcat 220902_SN1126_A_L001_AUHP-8_AdapterTrimmed_R1.fastq.gz | grep "^@" | more
   ```

3. **View sequences**:
   ```bash
   zcat 220902_SN1126_A_L001_AUHP-8_AdapterTrimmed_R1.fastq.gz | grep "^@" -A1 | grep -v "^@" | head
   ```

4. **Search for sequences with a specific barcode**:
   ```bash
   zcat 220902_SN1126_A_L001_AUHP-8_AdapterTrimmed_R1.fastq.gz | grep --color=always "ACACACACGAATTCCGTGAACCATCGAATCTTT" | more
   ```

5. **Count the number of sequences with a specific barcode**:
   ```bash
   zcat 220902_SN1126_A_L001_AUHP-8_AdapterTrimmed_R1.fastq.gz | grep --color=always "ACACACACGAATTCCGTGAACCATCGAATCTTT" | wc -l
   zcat 220902_SN1126_A_L001_AUHP-8_AdapterTrimmed_R1.fastq.gz | grep --color=always "GTGTACATGAATTCCGTGAACCATCGAATCTTT" | wc -l
   zcat 220902_SN1126_A_L001_AUHP-8_AdapterTrimmed_R1.fastq.gz | grep --color=always "ACACACACTCCTCCGCTTATTGATATGC" | wc -l
   ```

6. **Extract all barcodes with their counts**:
   ```bash
   zcat 220902_SN1126_A_L001_AUHP-8_AdapterTrimmed_R1.fastq.gz | grep "@" -A 1 | grep -v "^--" | grep -v "@" | cut -c 4-11 | sort | uniq -c | sort -n -r
   ```

7. **Check for Illumina adapters in sequences**:
   ```bash
   zcat 220902_SN1126_A_L001_AUHP-8_AdapterTrimmed_R1.fastq.gz | grep "AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC" | wc -l
   ```

8. **Extract the first 1 million reads**:
   ```bash
   zcat 220902_SN1126_A_L001_AUHP-8_AdapterTrimmed_R1.fastq.gz | head -n 4000000 | gzip > subset_220902_R1.fastq.gz
   zcat 220902_SN1126_A_L001_AUHP-8_AdapterTrimmed_R2.fastq.gz | head -n 4000000 | gzip > subset_220902_R2.fastq.gz
   # Verify the extraction
   zcat subset_220902_R1.fastq.gz | wc -l | awk '{print $1/4}'
   ```

### Check the Quality of Data Using FastQC

1. **Analyze a subset of data to speed up the process**:
   ```bash
   fastqc subset_220902_R1.fastq.gz
   # Open the resulting HTML file in a web browser
   # Repeat for the paired-end sequences
   fastqc subset_220902_R2.fastq.gz
   ```

2. **FastQC Report**:
   - **Per Base Sequence Quality**: High-quality scores (green) indicate good quality, while low scores (red) indicate poor quality.
   - **Per Tile Sequence Quality**: View of the quality scores across different tiles of the flow cell used during sequencing.
   - **Per Sequence Quality Scores**: Ideally, most sequences should have high average quality scores.
   - **Per Base Sequence Content**: Ideally, the lines should be relatively flat, indicating uniform base composition.
   - **Per Sequence GC Content**: Ideally, the distribution should be a normal distribution centered around the expected GC content.
   - **Per Base N Content**: Ideally, the proportion should be very low.
   - **Sequence Length Distribution**: Ideally, the sequences should be of uniform length.
   - **Sequence Duplication Levels**: High duplication levels are OK in DNA metabarcoding data due to the presence of abundant taxa.
   - **Overrepresented Sequences**: List of sequences that occur more frequently than expected.
   - **Adapter Content**: Ideally, there should be minimal adapter contamination.
   - **Kmer Content**: High enrichment at the beginning of sequences is OK due to the presence of barcodes.

---

[Previous Lesson](../lesson4/lesson4.md) | [Next Lesson](../lesson6/lesson6.md)
